File: apps/api/schemas/resume.py
from pydantic import BaseModel
from typing import List, Optional
from .personal_info import PersonalInfoIn
from .experience import ExperienceIn
from .education import EducationIn
from pydantic import Field as PydanticField
from .skill import SkillIn
from .project import ProjectIn
from .certification import CertificationIn

class ResumeIn(BaseModel):
    personal_info: PersonalInfoIn = PydanticField(alias="personalInfo")
    summary: Optional[str] = None
    skills: Optional[List[SkillIn]] = None
    experiences: Optional[List[ExperienceIn]] = None
    education: Optional[List[EducationIn]] = None
    projects: Optional[List[ProjectIn]] = None
    certifications: Optional[List[CertificationIn]] = None

File: apps/web/lib/types/resume.ts
export interface Experience {
  id: string;
  company: string;
  location: string;
  position: string;
  startDate: string;
  endDate: string;
  achievements: string[];
}

export interface Skill {
  id: string;
  skill: string;
  category: string;
}

export interface Education {
  id: string;
  school: string;
  degree: string;
  field: string;
  location: string;
  graduationDate: string;
}

export interface Project {
  id: string;
  name: string;
  highlights: string[];
  githubLink: string;
  websiteLink: string;
}

export interface Certification {
  id: string;
  title: string;
  issuer: string;
  date: string;
}

export interface BaseResume {
  personalInfo: {
    fullName: string;
    email: string;
    phone: string;
    location: string;
    linkedin?: string;
    github?: string;
  };
  summary: string;
  experiences: Experience[];
  education: Education[];
  skills: Skill[];
  projects?: Project[];
  certifications?: Certification[];
}

